adoc/chapters/programming_interface.adoc

```diff
@@ -355,94 +355,115 @@ are hidden from qualified and unqualified lookup.
 
-a@
-[source]
+_Effects:_ Moves from [code]#rhs# and assigns to this [code]#T# instance in
+accordance with the requirements set out above.
+
+'''
+
+.[apititle]#Destructor#
+[source,role=synopsis,id=api:reference-semantics-destructor]
 ----
 ~T()
 ----
-   a@ Destroys this SYCL [code]#T# instance in accordance with the
-      requirements set out in <<sec:reference-semantics>>. On destruction
-      of the last copy, may perform additional lifetime related operations
-      required for the underlying <<native-backend-object>> specified in
-      the <<backend>> specification document, if this SYCL [code]#T# instance
-      was originally constructed using one of the backend interoperability
-      [code]#make_*# functions specified in
-      <<sec:backend-interoperability-make>>.
-      See the relevant backend specification for details.
 
-|====
+_Effects:_ Destroys this [code]#T# instance in accordance with the requirements
+set out above.
 
-[[table.hiddenfriends.common.reference]]
-.Common hidden friend functions for reference semantics
-[width="100%",options="header",separator="@"]
-|====
-@ Hidden friend function @ Description
-a@
-[source]
+_Remarks:_ On destruction of the last copy, may perform additional
+lifetime-related operations required for the underlying
+<<native-backend-object>> specified in the <<backend>> specification document,
+if this [code]#T# instance was originally constructed using one of the backend
+interoperability [code]#make_*# functions specified in
+<<sec:backend-interoperability-make>>.
+See the relevant backend specification for details.
+
+'''
+
+[[sec:reference-semantics-hidden-friends]]
+==== Hidden friend functions
+
+This section describes the hidden friend functions that must be provided by all
+classes providing the common reference semantics.
+Each of these classes may provide further member functions, static member
+functions, etc., as documented in their respective sections.
+
+'''
+
+.[apititle]#Equality operator#
+[source,role=synopsis,id=api:reference-semantics-equality]
 ----
 bool operator==(const T& lhs, const T& rhs)
 ----
-   a@ Returns true if this LHS SYCL [code]#T# is equal to the RHS SYCL
-      [code]#T# in accordance with the requirements set out above,
-      otherwise returns false.
 
-a@
-[source]
+_Effects:_ Returns true if [code]#lhs# is equal to [code]#rhs# in accordance
+with the requirements set out above, otherwise returns false.
+
+'''
+
+.[apititle]#Inequality operator#
+[source,role=synopsis,id=api:reference-semantics-inequality]
 ----
 bool operator!=(const T& lhs, const T& rhs)
 ----
-   a@ Returns true if this LHS SYCL [code]#T# is not equal to the RHS
-      SYCL [code]#T# in accordance with the requirements set out above,
-      otherwise returns false.
-
-|====
+_Effects:_ Returns true if [code]#lhs# is not equal to [code]#rhs# in accordance
+with the requirements set out above, otherwise returns false.
```

Comment:
```suggestion
_Effects:_ Equivalent to [code]#return not (lhs == rhs);#.
```

This would prevent us from having to duplicate the wording about equality, if we make that change.